variables.go

```diff
@@ -137,15 +147,26 @@ func (r *reference) resolveEnv(cfg *Config, opts *options) (string, error) {
 
 func (r *reference) resolve(cfg *Config, opts *options) (value, error) {
 	v, err := r.resolveRef(cfg, opts)
-	if v != nil || err != nil {
+	if v != nil || (err != nil && (err != ErrCyclicReference && err != ErrMissing)) {
```

Comment:
hm... we have this logic twice. Another place is with `refDynValue`... At least for the check let's add a predicate, so we will have to update only one function if we need to adapt the check.